### PR TITLE
Add ApplyGuardrailScope for security guardrail evaluation tracing

### PR DESCRIPTION
**Pull request by nikhilNava**
## Summary

Introduces a new OpenTelemetry tracing scope (\ApplyGuardrailScope\) that captures security guardrail evaluations as spans. This enables observability into content safety, policy enforcement, and risk assessment decisions made during agent operations.

Port of [microsoft/Agent365-dotnet#252](https://github.com/microsoft/Agent365-dotnet/pull/252).

### New Contracts
- **\GuardrailDetails\** — Immutable contract capturing guardian evaluation metadata: target type, decision, provider info, policy details, and content hash for forensic correlation.
- **\GuardrailFinding\** — Represents an individual risk finding with severity, category, and confidence score.
- **\GuardrailDecisionType\** — Enum for well-known guardian decisions (Allow, Audit, Deny, Modify, Warn).
- **\GuardrailRiskSeverity\** — String constants for risk severity levels (None, Low, Medium, High, Critical).
- **\GuardrailTargetType\** — Extensible type for guardrail targets (LlmInput, LlmOutput, ToolCall, etc.).

### New Tracing Scope
- **\ApplyGuardrailScope\** — Disposable OpenTelemetry scope following the same pattern as \InvokeAgentScope\, \InferenceScope\, and \ExecuteToolScope\. Supports recording findings post-creation and integrates with the existing span hierarchy.

### ETW Support
- **\ApplyGuardrailData\ / \ApplyGuardrailDataBuilder\** — DTO and builder for Event Tracing for Windows logging of guardrail events.
- Updated \IA365EtwLogger\ and \A365EtwLogger\ with \LogApplyGuardrail\ method.

### OpenTelemetry Constants
- Added \microsoft.security.*\ and \microsoft.guardian.*\ attribute keys to \OpenTelemetryConstants\.

### Base Infrastructure
- Added \AddEvent()\ helper to \OpenTelemetryScope\ for span events.
- Added \AddSdkAttributes()\ to \BaseDataBuilder\ (and called from all existing builders).

### Tests
- 12 unit tests covering scope creation, finding recording, attribute emission, and edge cases.

## Design Notes

- Guardian spans are children of the operation they protect (e.g., inference or tool execution spans).
- Multiple guardian spans may exist under a single operation when guardrails are chained.
- Follows the existing Core + Extensions pattern and disposable scope lifecycle.
- All 434 existing tests continue to pass.

### CODE REVIEW

File: src/Microsoft.OpenTelemetry/Agent365/Runtime/Tracing/Scopes/OpenTelemetryConstants.cs [hunk 1]
```diff
@@ -49,9 +49,17 @@ public enum OperationNames
...
+            OutputMessages,
+
+            [EnumMember(Value = "ApplyGuardrail")]
+            ApplyGuardrail
         }
 
+        /// <summary>
+        /// The operation name value for apply_guardrail spans.
+        /// </summary>
+        public const string ApplyGuardrailOperationName = "apply_guardrail";
```
[the remark this answers is not in the document]
Comment (nikhilNava) [reply src/Microsoft.OpenTelemetry/Agent365/Runtime/Tracing/Scopes/OpenTelemetryConstants.cs:61]: fixed

File: src/Microsoft.OpenTelemetry/Agent365/Runtime/DTOs/Builders/ApplyGuardrailDataBuilder.cs [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+        private static Dictionary<string, object?> BuildAttributes(
+            GuardrailDetails guardrailDetails,
+            AgentDetails agentDetails,
+            string conversationId,
+            Channel? channel,
+            CallerDetails? callerDetails,
+            IDictionary<string, object?>? extraAttributes = null)
+        {
+            var attributes = new Dictionary<string, object?>();
+
```
[the remark this answers is not in the document]
Comment (nikhilNava) [reply src/Microsoft.OpenTelemetry/Agent365/Runtime/DTOs/Builders/ApplyGuardrailDataBuilder.cs:60]: fixed

File: src/Microsoft.OpenTelemetry/.publicApi/PublicAPI.Unshipped.txt [hunk 3]
```diff
@@ -29,3 +29,84 @@ Microsoft.Agents.A365.Observability.Runtime.Tracing.Exporters.TokenResolverConte
 Microsoft.Agents.A365.Observability.Runtime.Tracing.Exporters.TokenResolverContext.TenantId.get -> string!
 Microsoft.Agents.A365.Observability.Runtime.Tracing.Exporters.Agent365ExporterOptions.ContextualTokenResolver.get -> Microsoft.Agents.A365.Observability.Runtime.Tracing.Exporters.AsyncContextualTokenResolver?
 Microsoft.Agents.A365.Observability.Runtime.Tracing.Exporters.Agent365ExporterOptions.ContextualTokenResolver.set -> void
+Microsoft.Agents.A365.Observability.Runtime.Tracing.Contracts.GuardrailDecisionType
```
Comment (rajkumar-rangaraj) [inline src/Microsoft.OpenTelemetry/.publicApi/PublicAPI.Unshipped.txt:32]: Do we need these many public api's for this feature?

**Review by fpfp100 — APPROVED** [review]

File: src/Microsoft.OpenTelemetry/.publicApi/PublicAPI.Unshipped.txt (hunk 3 above)
Comment (nikhilNava) [reply src/Microsoft.OpenTelemetry/.publicApi/PublicAPI.Unshipped.txt:32]: It is mostly contracts for ETW and scopes for A365 which is why the larger footprint

**Review by juliomenendez — APPROVED** [review]

**Review by rajkumar-rangaraj — APPROVED** [review]
